### PR TITLE
Rust: support packaging and publishing

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -536,6 +536,11 @@ jobs:
       - run:
           name: Test
           command: cargo test
+      - run:
+          name: Package
+          command: |
```
Comment (chfast) [inline circle.yml:541]: Fancy way of doing this.
```suggestion
          working_dir: bindings/rust
          command: cargo package
```

File: bindings/rust/build.rs [hunk 2]
```diff
@@ -11,8 +11,22 @@ use std::env;
 use std::path::PathBuf;
 
 fn main() {
-    // This is the root directory.
-    let src = "../../";
```
Comment (axic) [inline bindings/rust/build.rs:15]: For some reason `cargo package` copies over symlinks to files with an updated path (that's why this works on evmc, where we only need the header which is symlinked), but symlinking the root does not work (e.g.` bindings/rust/fizzy -> ../../`).

File: circle.yml (hunk 1 above)
Comment (axic) [reply circle.yml:541]: How can I leave a comment in that case? Just as a yaml comment?

Comment (chfast) [reply circle.yml]: Yes. YAML commands are better IMHO.

Comment (axic) [reply circle.yml]: Is `~/bindings/rust` the current path? The suggested one does not work.

File: circle.yml [hunk 3]
```diff
@@ -539,7 +539,7 @@ jobs:
       - run:
           name: Package
           # The package must be run within the actual crate and not in the workspace.
-          working_dir: bindings/rust
+          working_dir: ~/bindings/rust
```
Comment (chfast) [inline circle.yml:542]:
```suggestion
          working_dir: ~/project/bindings/rust
```
if absolute path is required.

Comment (chfast) [reply circle.yml]: Ok, it should have been `working_directory`.

Comment (axic) [reply circle.yml]: Neither `bindings/rust`, `~/bindings/rust`, or `~/project/bindings/rust` works. Will check `pwd` because this is confusing.

**Review by chfast — APPROVED** [review]

**Review by gumb0 — APPROVED** [review]

**Pull request merged**